test/src/tests-store.ts

```diff
@@ -288,6 +293,202 @@ describe('Store', () => {
+        it('should return the firstSeenTime from the store', () => {
+            debugger;
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+            store.persistenceData[testMPID] = {
+                fst: 12345,
+            };
+            expect(store.getFirstSeenTime(testMPID)).to.equal(12345);
+        });
+
+        it('should return null if no firstSeenTime is found', () => {
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+            expect(store.getFirstSeenTime(testMPID)).to.equal(null);
+        });
+    });
+
+    describe('#setFirstSeenTime', () => {
+        it('should set the firstSeenTime in the store', () => {
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+
+            store.setFirstSeenTime(testMPID, 12345);
+            expect(store.persistenceData[testMPID].fst).to.equal(12345);
+        });
+
+        it('should return undefined if mpid is null', () => {
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+
+            expect(store.setFirstSeenTime(null, 12345)).to.equal(undefined);
+        });
+
+        it('should set the firstSeenTime in persistence', () => {
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+
+            store.setFirstSeenTime(testMPID, 12345);
+            const fromPersistence = window.mParticle
+                .getInstance()
+                ._Persistence.getPersistence();
+
+            expect(fromPersistence[testMPID]).to.be.ok;
+            expect(fromPersistence[testMPID].fst).to.be.ok;
+            expect(fromPersistence[testMPID].fst).to.equal(12345);
+        });
+
+        it('should override persistence with store values', () => {
+            const persistenceValue = JSON.stringify({
+                gs: {
+                    sid: 'sid',
+                    les: new Date().getTime(),
+                },
+                testMPID: {
+                    fst: 12345,
+                },
+                cu: testMPID,
+            });
+
+            localStorage.setItem(workspaceCookieName, persistenceValue);
+
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+
+            store.setFirstSeenTime(testMPID, 54321);
+            const fromPersistence = window.mParticle
+                .getInstance()
+                ._Persistence.getPersistence();
+
+            expect(fromPersistence[testMPID]).to.be.ok;
+            expect(fromPersistence[testMPID].fst).to.be.ok;
+            expect(fromPersistence[testMPID].fst).to.equal(54321);
```

Comment:
If the test is to see that it is overriding, maybe we test that the sid is null?  As opposed to the store extended what is currently in persistence?  It's a weird test right now but technically that's "overriding" what is in persistence.

Reply:
or I realize you can set an sid of `foo-sid` directly on the store before calling setFirstSeenTime and then check that the sid is now `foo-sid`